packages/react-instantsearch-core/src/connectors/connectVoiceSearch.js

```diff
@@ -0,0 +1,159 @@
+import PropTypes from 'prop-types';
+import createConnector from '../core/createConnector';
+import {
+  cleanUpValue,
+  refineValue,
+  getCurrentRefinementValue,
+  getIndexId,
+} from '../core/indexUtils';
+
+function getId() {
+  return 'query';
+}
+
+function getAdditionalId() {
+  return 'additionalVoiceParameters';
+}
+
+function getCurrentRefinementQuery(props, searchState, context) {
+  const id = getId();
+  const currentRefinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    ''
+  );
+
+  if (currentRefinement) {
+    return currentRefinement;
+  }
+  return '';
+}
+
+function getCurrentRefinementAdditional(props, searchState, context) {
+  const id = getAdditionalId();
+  const currentRefinement = getCurrentRefinementValue(
+    props,
+    searchState,
+    context,
+    id,
+    ''
+  );
+
+  if (currentRefinement) {
+    return currentRefinement;
+  }
+  return {};
+}
+
+function refine(props, searchState, nextRefinement, context) {
+  const id = getId();
+  const voiceParams = getAdditionalId();
+  const queryLanguages = props.language
+    ? { queryLanguages: [props.language.split('-')[0]] }
+    : {};
+  const additionalQueryParameters =
+    typeof props.additionalQueryParameters === 'function'
+      ? {
+          ignorePlurals: true,
+          removeStopWords: true,
+          optionalWords: nextRefinement,
+          ...props.additionalQueryParameters({ query: nextRefinement }),
+        }
+      : {};
+  const nextValue = {
+    [id]: nextRefinement,
+    [voiceParams]: {
```

Comment:
I think this part isn't tested yet (the id itself)

Reply:
here it is 8ef899d